build.gradle

```
@@ -139,7 +139,59 @@ allprojects {
     }
 }
 
-task tar(type: Tar) {
+tasks.register('dist') {
+    subprojects.forEach { subProject ->
+        dependsOn("${subProject.path}:jar")
+    }
```

I tried using the basic `dependsOn` statement, but then the plugin jar packages is incomplete.

```suggestion
    dependsOn("jar")
```

The existing writting is more complicated than the old one, is this the simplest writting yet?

---

Unless I am mistaken `dependsOn("jar")` means that the `:dist` task from the root project will depend on the `:jar` task from the root project. Here I accentuate that it depends on **all** tasks of type `jar`, e.g. `:eventmesh-common:jar`.

---

>Here I accentuate that it depends on **all** tasks of type `jar`, e.g. `:eventmesh-common:jar`.

Yeah I understand your code's meaning.

>Unless I am mistaken `dependsOn("jar")` means that the `:dist` task from the root project will depend on the `:jar` task from the root project.

The original `dist` task did depend on the `:jar` task from the root project and it works. I think it should be the same with your way. Is it because the use of the `task.register` writting makes it impossible to depend on the `jar` task from the root project anymore?

---

In the current build script on `master`, the `dist` task is declared inside `allprojects`, therefore each sub project has its own `dist` task that depends on its own `jar` task.

After the rewrite, there is only one `dist` task. Should I split the `dist` task again among projects?